keyboards/kingly_keys/ave/ortho/keymaps/default/keymap.c

```diff
@@ -0,0 +1,212 @@
+//  * └────┘ └────┘                                       └────┴────┴────┘
+//  * ┌────┐ ┌────┬────┬────┬────┬────┬────┬────┬────┬────┬────┬────┬────┐
+//  * │    │ │    │    │    │    │    │    │    │    │    │    │    │    │
+//  * ├────┤ ├────┼────┼────┼────┼────┼────┼────┼────┼────┼────┼────┼────┤
+//  * │    │ │    │    │    │    │    │    │    │    │    │    │    │    │
+//  * ├────┤ ├────┼────┼────┼────┼────┼────┼────┼────┼────┼────┼────┼────┤
+//  * │    │ │    │    │    │    │    │    │    │    │    │    │    │    │
+//  * ├────┤ ├────┼────┼────┼────┼────┼────┴────┼────┼────┼────┼────┼────┤
+//  * │    │ │    │    │    │    │    │    │    │    │    │    │    │    │
+//  * └────┘ └────┴────┴────┴────┴────┴─────────┴────┴────┴────┴────┴────┘
+//  */
+//  [<LAYER>] = LAYOUT_ortho_all(
+//      _______,      _______,                                                                                           _______,   _______,   _______,
+//      _______,      _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,
+//      _______,      _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,
+//      _______,      _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,
+//      _______,      _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______,   _______
+//  )
+
+
+
+// Encoder Customization: (*Order-of-Keycode Specific)
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+}
+
+
+
+// RGB Indicator Customization: (100% stolen from Jetpacktuxedo)
+void keyboard_post_init_user(void) {
+  #ifdef RGBLIGHT_ENABLE
+// Set up RGB effects on _only_ the first LED (index 0):
+        rgblight_set_effect_range(0, 1);
+// Set LED effects to breathing mode in a "terminal-green" type color:
+        rgblight_sethsv_noeeprom(50, 255, 100);
+        rgblight_mode_noeeprom(RGBLIGHT_EFFECT_BREATHING + 2);
+// Init the second LED to a static color:
+        setrgb(225, 185, 0, (LED_TYPE *)&led[1]);
+    rgblight_set();
+  #endif // RGBLIGHT_ENABLE
+}
+
+// RGB Indicator Customization: (cont.)
+layer_state_t layer_state_set_user(layer_state_t state){
+    #ifdef RGBLIGHT_ENABLE
+        uint8_t led1r = 0; uint8_t led1g = 0; uint8_t led1b = 0;
+            if (layer_state_cmp(state, 1)) {
+                led1b = 255;
+            }
+            if (layer_state_cmp(state, 3)) {
+                led1r = 200;
+            }
+            setrgb(led1r, led1g, led1b, (LED_TYPE *)&led[1]);
+        rgblight_set();
+    #endif //RGBLIGHT_ENABLE
+  return state;
+}
```

Comment:
Actually, I think that most (if not all) of this can be replaced by the rgb light layers feature. 

https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers

This may be better, since it won't get overwritten by effects.